pkg/accelerator/accelDeviceProvider.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accelerator
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/intel/sriov-network-device-plugin/pkg/types"
+	"github.com/jaypipes/ghw"
+)
+
+type accelDeviceProvider struct {
+	deviceList []types.AccelDevice
+	rFactory   types.ResourceFactory
```

Comment:
not related to the commit, but the factory could really be a singleton instead of referencing to it all over the place as
it creates instances of interfaces used in all layers of the code, requiring to store it and pass it down.

Reply:
factory is already implemented as singleton. It doesn't make any difference really whether we use the constructor everywhere to get the instance or pass it as an argument. Its one single instance everywhere.

Reply:
yah, it just needlessly holds an extra attribute ...

Reply:
I don't understand it. Even if we create an instance of the factory here(instead of getting this as an argument in its constructor NewAccelDeviceProvider() ) we still need an attribute for it's reference, no?

Reply:
What i mean is that every component that requires to call factory will do (for example):
`factory.Factory.<factory method>`
or
`factory.Get().<factory method>`

It would require to modify `ResourceFactory.GetResourceServer` to get the endpoint prefix/suffix and pluginWatch  (so we dont retain parameters in `resourceFactory` struct)

Then what you get in return is that you dont need to store a reference to it (e.g in netDeviceProvider, accelDeviceProvider)  and you don't need to pass it in their New methods, it just simplifies the code imo.

In regards to functionality it would be the same.

Reply:
To be clear, thats not something i request to be changed here, its just an observation.
ResourceFactory existed before this change.

Reply:
Ok, maybe some improvement for future.